catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -422,6 +469,12 @@ public DeleteResponse delete(DeleteRequest deleteRequest) throws IngestException
                     identifiers.size());
             deleteListOfMetacards(deletedMetacards, identifierPaged, attributeName);
         }
+        addPendingNrtDeletedMetacards(deletedMetacards, identifiers);
```

Comment:
So the flow i've walked thru and have in my head for this method is:
```
If 
    the number of actually deleted metacards is less than the number they asked to be deleted AND any of the deleteRequested ids are currently in the pending cache,
then 
    get any deleteRequested metacards from the pending cache, 
        then remove any of the metacards from that that are in the deleted metacards response (ones that were actually deleted), 
whatever is remaining is things that were requested for deletion, but were only in the pending cache but not the actual delete response.```

So my question is, why would something be in the cache but not come back as in the list of deleted metacards?